addons/google_calendar/models/google_sync.py

```diff
@@ -224,6 +225,15 @@ def _get_records_to_sync(self, full_sync=False):
     def _write_from_google(self, gevent, vals):
         self.write(vals)
 
+    def _get_calendar_user(self):
+        """ Return the responsible user of an event. If the responsible user doesn't exist or hasn't synced with google,
+        it will return the current user.
+        """
+        calendar_user = self.env.user
+        if self.env.user != self.user_id and self.user_id and self.user_id.google_calendar_rtoken:
+            calendar_user = self.user_id
+        return calendar_user
+
```

Comment:
Because of the way requests are done, you could user self.ensure_one in _get_calendar_user.